core/consensus/xpoa/xpoa.go

```diff
@@ -351,8 +351,14 @@ Again:
 // getCurrentValidates return current validates from xmodel
 // 注意：当查不到的时候或者一个候选人都没查到则默认取初始化的值
 // TODO: zq needs to be optimized in future because
-func (xpoa *XPoa) getCurrentValidates() ([]*cons_base.CandidateInfo, int64, int64, error) {
-	contractRes, confirmedTime, confirmedHeight, err := xpoa.utxoVM.SystemCall(xpoa.xpoaConf.contractName, xpoa.xpoaConf.methodName, nil, true)
+func (xpoa *XPoa) getCurrentValidates(curHeight int64) ([]*cons_base.CandidateInfo, int64, int64, error) {
+	preBlockId, err := xpoa.ledger.QueryBlockByHeight(curHeight)
+	xmod := xpoa.utxoVM.GetXModel()
+	reader, err := xmod.CreateSnapshot(preBlockId.GetBlockid())
+	if err != nil {
+		reader = xmod
```

Comment:
If set xmod, it may cause a bug. whether the default value should be returned?

Reply:
Default configuration might cause unavoidable mistake since default proposers might have been deleted while the system continue to use them.

Reply:
Finally, it's seem to be more reasonable while return error with nil value, where users can handle the specific error under no risk.